partiql-types/src/main/kotlin/org/partiql/value/io/PartiQLValueTextWriter.kt

```diff
@@ -192,6 +201,74 @@ public class PartiQLValueTextWriter(
             }
         }
 
+        override fun visitDate(v: DateValue, format: Format?) = v.toString(format) {
+            when (val value = v.value) {
+                null -> "null" // null.date
+                else -> sqlString(value)
+            }
+        }
+
+        private fun padZeros(v: Int, totalDigits: Int): String = String.format("%0${totalDigits}d", v)
+
+        private fun sqlString(d: Date): String {
+            val yyyy = padZeros(d.year, 4)
+            val mm = padZeros(d.month, 2)
+            val dd = padZeros(d.day, 2)
+            return "DATE '$yyyy-$mm-$dd'"
+        }
+
+        override fun visitTime(v: TimeValue, format: Format?) = v.toString(format) {
+            when (val value = v.value) {
+                null -> "null" // null.time
+                else -> sqlString(value)
+            }
+        }
+
+        private fun sqlString(tz: TimeZone?): String {
+            return when (tz) {
+                null -> ""
+                is TimeZone.UnknownTimeZone -> "-00:00"
+                is TimeZone.UtcOffset -> {
+                    val sign = if (tz.totalOffsetMinutes < 0) {
+                        "-"
+                    } else {
+                        "+"
+                    }
+                    val hh = padZeros(abs(tz.tzHour), 2)
+                    val mm = padZeros(abs(tz.tzMinute), 2)
+                    "$sign$hh:$mm"
+                }
+            }
+        }
+
+        private fun sqlString(t: Time): String {
+            val hh = padZeros(t.hour, 2)
+            val mm = padZeros(t.minute, 2)
+            val ss = padZeros(t.decimalSecond.toInt(), 2)
+            val frac = t.decimalSecond.remainder(BigDecimal.ONE).toString().substring(1) // drop leading 0
+            val timeZone = sqlString(t.timeZone)
+            return "TIME '$hh:$mm:$ss$frac$timeZone'"
+        }
+
+        override fun visitTimestamp(v: TimestampValue, format: Format?) = v.toString(format) {
+            when (val value = v.value) {
+                null -> "null" // null.timestamp
+                else -> sqlString(value)
+            }
+        }
+
+        private fun sqlString(t: Timestamp): String {
+            val yyyy = padZeros(t.year, 4)
+            val mon = padZeros(t.month, 2)
+            val dd = padZeros(t.day, 2)
+            val hh = padZeros(t.hour, 2)
+            val min = padZeros(t.minute, 2)
+            val ss = padZeros(t.decimalSecond.toInt(), 2)
+            val frac = t.decimalSecond.remainder(BigDecimal.ONE).toString().substring(1) // drop leading 0
```

Comment:
What is the reason to separate the big decimal in term of second and fraction second in print?

Reply:
I broke down the big decimal representing the seconds into full seconds and fractional to keep the full seconds portion of the string two characters.

E.g. if the seconds was `9.123`, I want to ensure the resulting string of the seconds was `09.123`. Perhaps I can simplify this logic further by just casing if the seconds is less than 10.

Reply:
Possible simplification:

```
val ss = if (t.decimalSecond < BigDecimal.TEN) {
    "0${t.decimalSecond}"
} else {
    t.decimalSecond.toString()
}
```

Reply:
Oh I see. Yeah I missed the padding zero part. Thanks.